### PR TITLE
Implement room finder api definition

### PR DESCRIPTION
**Pull request by joschahenningsen**

### CODE REVIEW

**Comment by joschahenningsen** [comment]
I suggest [bleve](http://blevesearch.com) for an improved searching experience. This snippet demonstrates its usage: 
https://gist.github.com/joschahenningsen/9dd5c3d1c4380ab990bce8de80a8552e

With their support for fuzzy searching, a misspelled query like `tentomaks garching` is still able to produce the expected result. Also it doesn't matter, which attribute of a room is searched for (e.g. id, code, name, description...)

Using bleve doesn't require much memory, just a few megabytes of disk space.

**Comment by joschahenningsen** [comment]
Feel free to use https://editor.swagger.io/ -> File -> Import URL -> `https://raw.githubusercontent.com/TUM-Dev/Campus-Backend/feat/roomfinder/api/gen/openapiv2/CampusService.swagger.json` to visualize this definition, I think it should be quite close to the existing one.

File: api/CampusService.proto [hunk 1]
```diff
@@ -24,6 +24,117 @@ service Campus {
       response_body: "sources"
     };
   }
+
+  rpc SearchRooms (SearchRoomsRequest) returns (SearchRoomsReply) {
+    option (google.api.http) = {
+      get: "/roomfinder/room/search/{query}",
```
Comment (kordianbruck) [inline api/CampusService.proto:30]: /roomfinder/search/

Use POST here to pass the query please.

File: api/CampusService.proto [hunk 2]
```diff
@@ -24,6 +24,117 @@ service Campus {
...
+  }
+
+  rpc GetLocations (GetLocationsRequest) returns (GetLocationsReply) {
+    option (google.api.http) = {
+      get: "/locations/{location}",
+      response_body: "locations"
+    };
+  }
+
+  rpc GetAvailableMaps (GetAvailableMapsRequest) returns (GetAvailableMapsReply) {
```
Comment (kordianbruck) [inline api/CampusService.proto:42]: GetRoomMaps

File: api/CampusService.proto [hunk 3]
```diff
@@ -24,6 +24,117 @@ service Campus {
...
+    option (google.api.http) = {
+      get: "/roomfinder/room/availableMaps/{arch_id}",
```
Comment (kordianbruck) [inline api/CampusService.proto:44]: /roomfinder/maps/

With Arch as POST parameter

File: api/CampusService.proto [hunk 4]
```diff
@@ -24,6 +24,117 @@ service Campus {
...
   }
+
+  rpc SearchRooms (SearchRoomsRequest) returns (SearchRoomsReply) {
+    option (google.api.http) = {
+      get: "/roomfinder/room/search/{query}",
+      response_body: "rooms"
+    };
+  }
+
+  rpc GetLocations (GetLocationsRequest) returns (GetLocationsReply) {
```
Comment (kordianbruck) [inline api/CampusService.proto:35]: Add a comment that this is university campus locations/buildings

File: api/CampusService.proto [hunk 5]
```diff
@@ -24,6 +24,117 @@ service Campus {
...
+  }
+
+  rpc GetAvailableMaps (GetAvailableMapsRequest) returns (GetAvailableMapsReply) {
+    option (google.api.http) = {
+      get: "/roomfinder/room/availableMaps/{arch_id}",
+      response_body: "maps"
+    };
+  }
+
+  rpc GetCoordinates (GetCoordinatesRequest) returns (GetCoordinatesReply) {
```
Comment (kordianbruck) [inline api/CampusService.proto:49]: GetRoomCoordinates

File: api/CampusService.proto [hunk 6]
```diff
@@ -24,6 +24,117 @@ service Campus {
...
+    option (google.api.http) = {
+      get: "/roomfinder/room/coordinates/{arch_id}",
```
Comment (kordianbruck) [inline api/CampusService.proto:51]: /roomfinder/coordinates/

With Arch as POST parameter

File: api/CampusService.proto [hunk 7]
```diff
@@ -24,6 +24,117 @@ service Campus {
...
+    option (google.api.http) = {
+      get: "/roomfinder/room/scheduleById/{room_id}/{start}/{end}",
+      response_body: "events"
+    };
+  }
+}
+
+message GetRoomScheduleRequest {
+  int32 room_id = 1;
+  string start = 2; // yyyyMMdd
```
Comment (kordianbruck) [inline api/CampusService.proto:65]: This should definately be a datetime type

File: api/CampusService.proto [hunk 8]
```diff
@@ -24,6 +24,117 @@ service Campus {
...
+  rpc GetRoomSchedule (GetRoomScheduleRequest) returns (GetRoomScheduleReply) {
+    option (google.api.http) = {
+      get: "/roomfinder/room/scheduleById/{room_id}/{start}/{end}",
+      response_body: "events"
+    };
+  }
+}
+
+message GetRoomScheduleRequest {
+  int32 room_id = 1;
```
Comment (kordianbruck) [inline api/CampusService.proto:64]: Do we need _id here?

File: api/CampusService.proto (hunk 7 above)
Comment (kordianbruck) [reply api/CampusService.proto:65]: e.g.: google.protobuf.Timestamp

File: api/CampusService.proto [hunk 9]
```diff
@@ -24,6 +24,117 @@ service Campus {
...
+  string start = 2; // yyyyMMdd
+  string end = 3; // yyyyMMdd
+}
+
+message GetRoomScheduleReply {
+  message RoomScheduleEvent {
+    google.protobuf.Timestamp start = 1;
+    google.protobuf.Timestamp end = 2;
+    string title = 3;
+    string event_id = 4;
```
Comment (kordianbruck) [inline api/CampusService.proto:74]: Is this really a string?

Comment (joschahenningsen) [reply api/CampusService.proto:74]: int in sql, string in the current API :crossed_swords:

**Review by kordianbruck — APPROVED** [review]

**Pull request merged**